infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/CommonPVCStrategy.java

```diff
@@ -258,4 +272,31 @@ private PersistentVolumeClaim replacePVCsWithCommon(
         .filter(subpath -> !isNullOrEmpty(subpath))
         .collect(Collectors.toSet());
   }
+
+  private void deleteCommonPVC(Workspace workspace) throws InfrastructureException {
+    factory.get(workspace).persistentVolumeClaims().delete(configuredPVCName);
+  }
+
+  /**
+   * @return true, if a given user has no workspaces, false otherwise (or if the subject is
+   *     anonymous)
+   * @throws RuntimeException
+   */
+  private boolean userHasNoWorkspaces() {
+    Subject subject = EnvironmentContext.getCurrent().getSubject();
+    if (!subject.isAnonymous()) {
+      String userId = subject.getUserId();
+      try {
+        Page<WorkspaceImpl> workspaces = workspaceManager.getWorkspaces(userId, false, 1, 0);
+        if (workspaces.isEmpty()) {
+          log.debug("User '{}' has no more workspaces left", userId);
+          return true;
+        }
+      } catch (ServerException e) {
+        // should never happen
+        throw new RuntimeException(e.getLocalizedMessage(), e);
```

Comment:
Just wonder. Maybe it's better to wrap it in `InfrastructureException` wdyt?

Reply:
sounds good, fixed